server/jobs/job_store.go

```diff
@@ -132,15 +160,13 @@ func (j *LayeredJobStore) SetJobCompleteStatus(jobID string, status JobStatus) e
 	job := j.jobs[jobID]
 	job.Status = Complete
 
-	// Persist to storage backend
-	ok, err := j.storageBackend.Write(jobID, job.Output)
-	if err != nil {
-		return errors.Wrapf(err, "error persisting job: %s", jobID)
-	}
-
-	// Clear output buffers if successfully persisted
-	if ok {
+	// Persist to storage backend and return error if error other than StorageBackendNotConfigured
+	err := j.storageBackend.Write(jobID, job)
+	if err == nil {
+		// Only remove from memory if logs are persisted successfully
 		delete(j.jobs, jobID)
+	} else if !errors.Is(err, &StorageBackendNotConfigured{}) {
```

Comment:
You wouldn't need this error check if you were using an interface with different implementations.

Reply:
Decided to use a two value return statement in the Write() method. So, we will no longer have to look for custom errors to decide if we want to delete the job from memory.